### PR TITLE
Copy orchestrator implementation from Celestia-app

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview
Closes https://github.com/celestiaorg/orchestrator-relayer/issues/11

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+type I interface {
+	Start(ctx context.Context)
+	StartNewEventsListener(ctx context.Context, queue chan<- uint64, signalChan <-chan struct{}) error
+	EnqueueMissingEvents(ctx context.Context, queue chan<- uint64, signalChan <-chan struct{}) error
+	ProcessNonces(ctx context.Context, noncesQueue <-chan uint64, signalChan chan<- struct{}) error
+	Process(ctx context.Context, nonce uint64) error
+	ProcessValsetEvent(ctx context.Context, valset celestiatypes.Valset) error
+	ProcessDataCommitmentEvent(ctx context.Context, dc celestiatypes.DataCommitment) error
+}
```
Comment (evan-forbes) [inline orchestrator/orchestrator.go:36]: naming an interface `I` feels very non-standard lol, but I think I see where this is coming from given that everything is named orchestrator. 

While we certainly don't have to handle in this PR, I think re-evaluating this, and other, interfaces would prove valuable . IIRC, we originally added the interface to make some unit tests easier, but then changes the design. if that's the case, then we should think about creating a different mock that can do something very similar. In general, I'm typically skeptical of large interfaces, especially when we only plan on supporting a single implementation.

speaking of which, do we also not have any unit tests for these? if not, mind creating an issue or handling here? thanks!

**Review by evan-forbes — COMMENTED** [review]
good start!

File: orchestrator/orchestrator.go (hunk 1 above)
Comment (SweeXordious) [reply orchestrator/orchestrator.go:36]: > then we should think about creating a different mock that can do something very similar

Do you have some suggestion or an example of using different mocks without having to create a general interface? Would love to take a look at it and improve on this

>  do we also not have any unit tests for these? 

Good catch https://github.com/celestiaorg/orchestrator-relayer/issues/51

Comment (evan-forbes) [reply orchestrator/orchestrator.go:36]: well, we could use generated mocks, but I'm not the biggest fan of those either, or mocking when not absolutely necessary. One thing _might_ be possible is to only mock the app queries.

Comment (SweeXordious) [reply orchestrator/orchestrator.go:36]: > We could also use a testnode

I like this, created issue: https://github.com/celestiaorg/orchestrator-relayer/issues/55

So, we just leave the current interface as it is for now? until we have a testnode?

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**